arangod/Cache/Table.cpp

```diff
@@ -34,7 +34,11 @@ using namespace arangodb::cache;
 const uint32_t Table::minLogSize = 8;
 const uint32_t Table::maxLogSize = 32;
 
-Table::GenericBucket::GenericBucket() : _state{}, _padding{0} {}
+Table::GenericBucket::GenericBucket() : _state{} {
+  for (size_t i = 0; i < paddingSize; i++) {
+    _padding[i] = static_cast<uint8_t>(0);
```

Comment:
This seems unnecessary since `_padding` is only defined to ensure that GenericBucket has the correct size, but is otherwise unused. However, if you really want to explicitly initialize it to zero, you can avoid the loop and simply do it in the initializer list (i.e., `_padding{}`).

Reply:
Currently the `GenericBucket` is used for some type-blind handling of the tables. The actual tables are just allocated with large memory chunks, which can later be used as arrays of `PlainBucket` or `TransactionalBucket` objects via casting, so the space does need to be zeroed out.

Is it the cleanest and most type-safe solution? Certainly not. Does it allow us to re-use large chunks of memory and avoid a bunch of extraneous large allocations? Yes.

Certainly open to reworking large parts of this caching code, and would agree that some of it definitely needs to be fixed. Currently though, that's how the allocations work and therefore it does need to be initialized.

Reply:
Alright, but in this case you also have to revert the changes in 38438bfbcca4c9f6b29c04a88b9148ad3e1ce242 to restore the clear function.

So if you are open to reworking this, I might tackle this in one of my next PRs. After a brief glance at the code I might have seen some potential issues (including race conditions), but I will have to take a closer look.

PS: I personally would still prefer initialization via initializer list 😉